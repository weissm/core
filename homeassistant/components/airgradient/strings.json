{
  "config": {
    "flow_title": "{model}",
    "step": {
      "user": {
        "data": {
          "host": "[%key:common::config_flow::data::host%]"
        },
        "data_description": {
          "host": "The hostname or IP address of the Airgradient device."
        }
      },
      "discovery_confirm": {
        "description": "Do you want to setup {model}?"
      }
    },
    "abort": {
      "already_configured": "[%key:common::config_flow::abort::already_configured_device%]",
      "invalid_version": "This firmware version is unsupported. Please upgrade the firmware of the device to at least version 3.1.1."
    },
    "error": {
      "cannot_connect": "[%key:common::config_flow::error::cannot_connect%]",
      "unknown": "[%key:common::config_flow::error::unknown%]"
    }
  },
  "entity": {
    "select": {
      "configuration_control": {
        "name": "Configuration source",
        "state": {
          "cloud": "Cloud",
          "local": "Local"
        }
      },
      "display_temperature_unit": {
        "name": "Display temperature unit",
        "state": {
          "c": "Celsius",
          "f": "Fahrenheit"
        }
      },
      "display_pm_standard": {
        "name": "Display PM standard",
        "state": {
          "ugm3": "µg/m³",
          "us_aqi": "US AQI"
        }
      },
      "led_bar_mode": {
        "name": "LED bar mode",
        "state": {
          "off": "Off",
          "co2": "Carbon dioxide",
          "pm": "Particulate matter"
        }
      }
    },
    "sensor": {
      "total_volatile_organic_component_index": {
        "name": "VOC index"
      },
      "nitrogen_index": {
        "name": "NOx index"
      },
      "pm003_count": {
        "name": "PM0.3"
      },
      "raw_total_volatile_organic_component": {
        "name": "Raw VOC"
      },
      "raw_nitrogen": {
        "name": "Raw NOx"
      }
    }
  },
  "exceptions": {
    "no_local_configuration": {
      "message": "Device should be configured with local configuration to be able to change settings."
    }
  }
}
