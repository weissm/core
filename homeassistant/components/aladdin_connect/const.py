"""Constants for the Aladdin Connect Genie integration."""

DOMAIN = "aladdin_connect"

OAUTH2_AUTHORIZE = "https://app.aladdinconnect.net/login.html"
OAUTH2_TOKEN = "https://twdvzuefzh.execute-api.us-east-2.amazonaws.com/v1/oauth2/token"
