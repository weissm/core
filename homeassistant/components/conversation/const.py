"""Const for conversation integration."""

DOMAIN = "conversation"
DEFAULT_EXPOSED_ATTRIBUTES = {"device_class"}
HOME_ASSISTANT_AGENT = "conversation.home_assistant"
OLD_HOME_ASSISTANT_AGENT = "homeassistant"

ATTR_TEXT = "text"
ATTR_LANGUAGE = "language"
ATTR_AGENT_ID = "agent_id"
ATTR_CONVERSATION_ID = "conversation_id"

SERVICE_PROCESS = "process"
SERVICE_RELOAD = "reload"
