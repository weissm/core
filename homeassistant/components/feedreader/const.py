"""Constants for RSS/Atom feeds."""

from datetime import timedelta

DOMAIN = "feedreader"

CONF_MAX_ENTRIES = "max_entries"
DEFAULT_MAX_ENTRIES = 20
DEFAULT_SCAN_INTERVAL = timedelta(hours=1)
