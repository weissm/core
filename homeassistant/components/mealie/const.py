"""Constants for the Mealie integration."""

import logging

DOMAIN = "mealie"

LOGGER = logging.getLogger(__package__)
