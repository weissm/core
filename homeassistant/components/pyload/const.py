"""Constants for the pyLoad integration."""

DOMAIN = "pyload"

DEFAULT_HOST = "localhost"
DEFAULT_NAME = "pyLoad"
DEFAULT_PORT = 8000
