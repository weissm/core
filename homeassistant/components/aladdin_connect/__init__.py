"""The Aladdin Connect Genie integration."""

from __future__ import annotations

from genie_partner_sdk.client import AladdinConnectClient

from homeassistant.config_entries import ConfigEntry
from homeassistant.const import Platform
from homeassistant.core import HomeAssistant
from homeassistant.helpers import device_registry as dr
from homeassistant.helpers.aiohttp_client import async_get_clientsession
from homeassistant.helpers.config_entry_oauth2_flow import (
    OAuth2Session,
    async_get_config_entry_implementation,
)

from .api import AsyncConfigEntryAuth
from .const import DOMAIN
from .coordinator import AladdinConnectCoordinator

PLATFORMS: list[Platform] = [Platform.COVER, Platform.SENSOR]

type AladdinConnectConfigEntry = ConfigEntry[AladdinConnectCoordinator]


async def async_setup_entry(
    hass: HomeAssistant, entry: AladdinConnectConfigEntry
) -> bool:
    """Set up Aladdin Connect Genie from a config entry."""
    implementation = await async_get_config_entry_implementation(hass, entry)

    session = OAuth2Session(hass, entry, implementation)
    auth = AsyncConfigEntryAuth(async_get_clientsession(hass), session)
    coordinator = AladdinConnectCoordinator(hass, AladdinConnectClient(auth))

    await coordinator.async_setup()
    await coordinator.async_config_entry_first_refresh()

    entry.runtime_data = coordinator

    await hass.config_entries.async_forward_entry_setups(entry, PLATFORMS)

    async_remove_stale_devices(hass, entry)

    return True


async def async_unload_entry(
    hass: HomeAssistant, entry: AladdinConnectConfigEntry
) -> bool:
    """Unload a config entry."""
    return await hass.config_entries.async_unload_platforms(entry, PLATFORMS)


async def async_migrate_entry(
    hass: HomeAssistant, config_entry: AladdinConnectConfigEntry
) -> bool:
    """Migrate old config."""
    if config_entry.version < 2:
        config_entry.async_start_reauth(hass)
        hass.config_entries.async_update_entry(
            config_entry,
            version=2,
            minor_version=1,
        )

    return True


def async_remove_stale_devices(
    hass: HomeAssistant, config_entry: AladdinConnectConfigEntry
) -> None:
    """Remove stale devices from device registry."""
    device_registry = dr.async_get(hass)
    device_entries = dr.async_entries_for_config_entry(
        device_registry, config_entry.entry_id
    )
    all_device_ids = {door.unique_id for door in config_entry.runtime_data.doors}

    for device_entry in device_entries:
        device_id: str | None = None

        for identifier in device_entry.identifiers:
            if identifier[0] == DOMAIN:
                device_id = identifier[1]
                break

        if device_id is None or device_id not in all_device_ids:
            # If device_id is None an invalid device entry was found for this config entry.
            # If the device_id is not in existing device ids it's a stale device entry.
            # Remove config entry from this device entry in either case.
            device_registry.async_update_device(
                device_entry.id, remove_config_entry_id=config_entry.entry_id
            )
