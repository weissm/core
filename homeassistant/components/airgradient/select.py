"""Support for AirGradient select entities."""

from collections.abc import Awaitable, Callable
from dataclasses import dataclass

from airgradient import AirGradientClient, Config
from airgradient.models import (
    ConfigurationControl,
    LedBarMode,
    PmStandard,
    TemperatureUnit,
)

from homeassistant.components.select import SelectEntity, SelectEntityDescription
from homeassistant.const import EntityCategory
from homeassistant.core import HomeAssistant
from homeassistant.exceptions import ServiceValidationError
from homeassistant.helpers.entity_platform import AddEntitiesCallback

from . import AirGradientConfigEntry
from .const import DOMAIN
from .coordinator import AirGradientConfigCoordinator
from .entity import AirGradientEntity

PM_STANDARD = {
    PmStandard.UGM3: "ugm3",
    PmStandard.USAQI: "us_aqi",
}
PM_STANDARD_REVERSE = {v: k for k, v in PM_STANDARD.items()}


@dataclass(frozen=True, kw_only=True)
class AirGradientSelectEntityDescription(SelectEntityDescription):
    """Describes AirGradient select entity."""

    value_fn: Callable[[Config], str | None]
    set_value_fn: Callable[[AirGradientClient, str], Awaitable[None]]
    requires_display: bool = False
    requires_led_bar: bool = False


CONFIG_CONTROL_ENTITY = AirGradientSelectEntityDescription(
    key="configuration_control",
    translation_key="configuration_control",
    options=[ConfigurationControl.CLOUD.value, ConfigurationControl.LOCAL.value],
    entity_category=EntityCategory.CONFIG,
    value_fn=lambda config: (
        config.configuration_control
        if config.configuration_control is not ConfigurationControl.NOT_INITIALIZED
        else None
    ),
    set_value_fn=lambda client, value: client.set_configuration_control(
        ConfigurationControl(value)
    ),
)

PROTECTED_SELECT_TYPES: tuple[AirGradientSelectEntityDescription, ...] = (
    AirGradientSelectEntityDescription(
        key="display_temperature_unit",
        translation_key="display_temperature_unit",
        options=[x.value for x in TemperatureUnit],
        entity_category=EntityCategory.CONFIG,
        value_fn=lambda config: config.temperature_unit,
        set_value_fn=lambda client, value: client.set_temperature_unit(
            TemperatureUnit(value)
        ),
        requires_display=True,
    ),
    AirGradientSelectEntityDescription(
        key="display_pm_standard",
        translation_key="display_pm_standard",
        options=list(PM_STANDARD_REVERSE),
        entity_category=EntityCategory.CONFIG,
        value_fn=lambda config: PM_STANDARD.get(config.pm_standard),
        set_value_fn=lambda client, value: client.set_pm_standard(
            PM_STANDARD_REVERSE[value]
        ),
        requires_display=True,
    ),
    AirGradientSelectEntityDescription(
        key="led_bar_mode",
        translation_key="led_bar_mode",
        options=[x.value for x in LedBarMode],
        entity_category=EntityCategory.CONFIG,
        value_fn=lambda config: config.led_bar_mode,
        set_value_fn=lambda client, value: client.set_led_bar_mode(LedBarMode(value)),
        requires_led_bar=True,
    ),
)


async def async_setup_entry(
    hass: HomeAssistant,
    entry: AirGradientConfigEntry,
    async_add_entities: AddEntitiesCallback,
) -> None:
    """Set up AirGradient select entities based on a config entry."""

    config_coordinator = entry.runtime_data.config
    measurement_coordinator = entry.runtime_data.measurement

    entities = [AirGradientSelect(config_coordinator, CONFIG_CONTROL_ENTITY)]

    entities.extend(
        AirGradientProtectedSelect(config_coordinator, description)
        for description in PROTECTED_SELECT_TYPES
        if (
            description.requires_display
            and measurement_coordinator.data.model.startswith("I")
        )
        or (description.requires_led_bar and "L" in measurement_coordinator.data.model)
    )

    async_add_entities(entities)


class AirGradientSelect(AirGradientEntity, SelectEntity):
    """Defines an AirGradient select entity."""

    entity_description: AirGradientSelectEntityDescription
    coordinator: AirGradientConfigCoordinator

    def __init__(
        self,
        coordinator: AirGradientConfigCoordinator,
        description: AirGradientSelectEntityDescription,
    ) -> None:
        """Initialize AirGradient select."""
        super().__init__(coordinator)
        self.entity_description = description
        self._attr_unique_id = f"{coordinator.serial_number}-{description.key}"

    @property
    def current_option(self) -> str | None:
        """Return the state of the select."""
        return self.entity_description.value_fn(self.coordinator.data)

    async def async_select_option(self, option: str) -> None:
        """Change the selected option."""
        await self.entity_description.set_value_fn(self.coordinator.client, option)
        await self.coordinator.async_request_refresh()


class AirGradientProtectedSelect(AirGradientSelect):
    """Defines a protected AirGradient select entity."""

    async def async_select_option(self, option: str) -> None:
        """Change the selected option."""
        if (
            self.coordinator.data.configuration_control
            is not ConfigurationControl.LOCAL
        ):
            raise ServiceValidationError(
                translation_domain=DOMAIN,
                translation_key="no_local_configuration",
            )
        await super().async_select_option(option)
