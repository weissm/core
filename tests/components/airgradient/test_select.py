"""Tests for the AirGradient select platform."""

from unittest.mock import AsyncMock, patch

from airgradient import ConfigurationControl, Measures
import pytest
from syrupy import SnapshotAssertion

from homeassistant.components.airgradient import DOMAIN
from homeassistant.components.select import (
    DOMAIN as SELECT_DOMAIN,
    SERVICE_SELECT_OPTION,
)
from homeassistant.const import ATTR_ENTITY_ID, ATTR_OPTION, Platform
from homeassistant.core import HomeAssistant
from homeassistant.exceptions import ServiceValidationError
from homeassistant.helpers import entity_registry as er

from . import setup_integration

from tests.common import MockConfigEntry, load_fixture, snapshot_platform


@pytest.mark.usefixtures("entity_registry_enabled_by_default")
async def test_all_entities(
    hass: HomeAssistant,
    snapshot: SnapshotAssertion,
    mock_airgradient_client: AsyncMock,
    mock_config_entry: MockConfigEntry,
    entity_registry: er.EntityRegistry,
) -> None:
    """Test all entities."""
    with patch("homeassistant.components.airgradient.PLATFORMS", [Platform.SELECT]):
        await setup_integration(hass, mock_config_entry)

    await snapshot_platform(hass, entity_registry, snapshot, mock_config_entry.entry_id)


@pytest.mark.usefixtures("entity_registry_enabled_by_default")
async def test_all_entities_outdoor(
    hass: HomeAssistant,
    snapshot: SnapshotAssertion,
    mock_airgradient_client: AsyncMock,
    mock_config_entry: MockConfigEntry,
    entity_registry: er.EntityRegistry,
) -> None:
    """Test all entities."""
    mock_airgradient_client.get_current_measures.return_value = Measures.from_json(
        load_fixture("current_measures_outdoor.json", DOMAIN)
    )
    with patch("homeassistant.components.airgradient.PLATFORMS", [Platform.SELECT]):
        await setup_integration(hass, mock_config_entry)

    await snapshot_platform(hass, entity_registry, snapshot, mock_config_entry.entry_id)


async def test_setting_value(
    hass: HomeAssistant,
    mock_airgradient_client: AsyncMock,
    mock_config_entry: MockConfigEntry,
) -> None:
    """Test setting value."""
    await setup_integration(hass, mock_config_entry)

    await hass.services.async_call(
        SELECT_DOMAIN,
        SERVICE_SELECT_OPTION,
        {
            ATTR_ENTITY_ID: "select.airgradient_configuration_source",
            ATTR_OPTION: "local",
        },
        blocking=True,
    )
    mock_airgradient_client.set_configuration_control.assert_called_once_with("local")
    assert mock_airgradient_client.get_config.call_count == 2


async def test_setting_protected_value(
    hass: HomeAssistant,
    mock_cloud_airgradient_client: AsyncMock,
    mock_config_entry: MockConfigEntry,
) -> None:
    """Test setting protected value."""
    await setup_integration(hass, mock_config_entry)

    with pytest.raises(ServiceValidationError):
        await hass.services.async_call(
            SELECT_DOMAIN,
            SERVICE_SELECT_OPTION,
            {
                ATTR_ENTITY_ID: "select.airgradient_display_temperature_unit",
                ATTR_OPTION: "c",
            },
            blocking=True,
        )
    mock_cloud_airgradient_client.set_temperature_unit.assert_not_called()

    mock_cloud_airgradient_client.get_config.return_value.configuration_control = (
        ConfigurationControl.LOCAL
    )

    await hass.services.async_call(
        SELECT_DOMAIN,
        SERVICE_SELECT_OPTION,
        {
            ATTR_ENTITY_ID: "select.airgradient_display_temperature_unit",
            ATTR_OPTION: "c",
        },
        blocking=True,
    )
    mock_cloud_airgradient_client.set_temperature_unit.assert_called_once_with("c")
