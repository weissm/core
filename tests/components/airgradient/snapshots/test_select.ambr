# serializer version: 1
# name: test_all_entities[select.airgradient_configuration_source-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': dict({
      'options': list([
        'cloud',
        'local',
      ]),
    }),
    'config_entry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'select',
    'entity_category': <EntityCategory.CONFIG: 'config'>,
    'entity_id': 'select.airgradient_configuration_source',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': None,
    'original_icon': None,
    'original_name': 'Configuration source',
    'platform': 'airgradient',
    'previous_unique_id': None,
    'supported_features': 0,
    'translation_key': 'configuration_control',
    'unique_id': '84fce612f5b8-configuration_control',
    'unit_of_measurement': None,
  })
# ---
# name: test_all_entities[select.airgradient_configuration_source-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'friendly_name': 'Airgradient Configuration source',
      'options': list([
        'cloud',
        'local',
      ]),
    }),
    'context': <ANY>,
    'entity_id': 'select.airgradient_configuration_source',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'local',
  })
# ---
# name: test_all_entities[select.airgradient_display_pm_standard-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': dict({
      'options': list([
        'ugm3',
        'us_aqi',
      ]),
    }),
    'config_entry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'select',
    'entity_category': <EntityCategory.CONFIG: 'config'>,
    'entity_id': 'select.airgradient_display_pm_standard',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': None,
    'original_icon': None,
    'original_name': 'Display PM standard',
    'platform': 'airgradient',
    'previous_unique_id': None,
    'supported_features': 0,
    'translation_key': 'display_pm_standard',
    'unique_id': '84fce612f5b8-display_pm_standard',
    'unit_of_measurement': None,
  })
# ---
# name: test_all_entities[select.airgradient_display_pm_standard-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'friendly_name': 'Airgradient Display PM standard',
      'options': list([
        'ugm3',
        'us_aqi',
      ]),
    }),
    'context': <ANY>,
    'entity_id': 'select.airgradient_display_pm_standard',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'ugm3',
  })
# ---
# name: test_all_entities[select.airgradient_display_temperature_unit-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': dict({
      'options': list([
        'c',
        'f',
      ]),
    }),
    'config_entry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'select',
    'entity_category': <EntityCategory.CONFIG: 'config'>,
    'entity_id': 'select.airgradient_display_temperature_unit',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': None,
    'original_icon': None,
    'original_name': 'Display temperature unit',
    'platform': 'airgradient',
    'previous_unique_id': None,
    'supported_features': 0,
    'translation_key': 'display_temperature_unit',
    'unique_id': '84fce612f5b8-display_temperature_unit',
    'unit_of_measurement': None,
  })
# ---
# name: test_all_entities[select.airgradient_display_temperature_unit-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'friendly_name': 'Airgradient Display temperature unit',
      'options': list([
        'c',
        'f',
      ]),
    }),
    'context': <ANY>,
    'entity_id': 'select.airgradient_display_temperature_unit',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'c',
  })
# ---
# name: test_all_entities[select.airgradient_led_bar_mode-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': dict({
      'options': list([
        'off',
        'co2',
        'pm',
      ]),
    }),
    'config_entry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'select',
    'entity_category': <EntityCategory.CONFIG: 'config'>,
    'entity_id': 'select.airgradient_led_bar_mode',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': None,
    'original_icon': None,
    'original_name': 'LED bar mode',
    'platform': 'airgradient',
    'previous_unique_id': None,
    'supported_features': 0,
    'translation_key': 'led_bar_mode',
    'unique_id': '84fce612f5b8-led_bar_mode',
    'unit_of_measurement': None,
  })
# ---
# name: test_all_entities[select.airgradient_led_bar_mode-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'friendly_name': 'Airgradient LED bar mode',
      'options': list([
        'off',
        'co2',
        'pm',
      ]),
    }),
    'context': <ANY>,
    'entity_id': 'select.airgradient_led_bar_mode',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'co2',
  })
# ---
# name: test_all_entities_outdoor[select.airgradient_configuration_source-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': dict({
      'options': list([
        'cloud',
        'local',
      ]),
    }),
    'config_entry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'select',
    'entity_category': <EntityCategory.CONFIG: 'config'>,
    'entity_id': 'select.airgradient_configuration_source',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': None,
    'original_icon': None,
    'original_name': 'Configuration source',
    'platform': 'airgradient',
    'previous_unique_id': None,
    'supported_features': 0,
    'translation_key': 'configuration_control',
    'unique_id': '84fce612f5b8-configuration_control',
    'unit_of_measurement': None,
  })
# ---
# name: test_all_entities_outdoor[select.airgradient_configuration_source-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'friendly_name': 'Airgradient Configuration source',
      'options': list([
        'cloud',
        'local',
      ]),
    }),
    'context': <ANY>,
    'entity_id': 'select.airgradient_configuration_source',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'local',
  })
# ---
